packages/flutter/lib/src/material/page.dart

```diff
@@ -88,12 +89,25 @@ class AnimationMean extends CompoundAnimation<double> {
 // is fully on-screen. 0.0 is fully off-screen to the right. 1.0 is off-screen
 // to the left.
 class _CupertinoTransitionCurve extends Curve {
-  _CupertinoTransitionCurve();
+  _CupertinoTransitionCurve(this.curve);
+
+  Curve curve;
 
   @override
   double transform(double t) {
-    if (t > 0.5)
-      return (t - 0.5) / 3.0 + 0.5;
+    // The range is divided in 2 halves: [0.0, 0.5] applies to the top-most
+    // route, while [0.5, 1.0] applies to the one underneath it. We scale both
+    // ranges to [0.0, 1.0] when applying transformations.
```

Comment:
Ah, so maybe it's not true :). Probably why it didn't make sense.

It's hard for me to reason about what should or would happen when there are multiple transitions happening at once, so I only focused on the simple case of a push or pop. My goal with this code was to have the topmost page have fast transition from all-the-way-to-the-right to onscreen, while the page underneath has a slower transition from onscreen to about-a-third-to-the-left. Probably clearer if you watch what the iOS transition is doing. But in the simple case, t=0.5 represents the case when the topmost page is fully on screen, and the transition is settled.